samples/cpp/benchmark_app/result_dump.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "result_dump.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include "samples/slog.hpp"
+
+void ResultDump::add_result(const std::string& input_image, const ov::TensorVector& outputs) {
+    // Compare with exist output, if it is different, save it to buffer and write it to binary file.
+    if (compare_output(input_image, outputs)) {
+        // Check if the number of outputs has reached the maximum value
+        // Stop saving if the number of different outputs exceeds the maximum
+        auto it = _different_outputs.find(input_image);
+        if (it != _different_outputs.end()) {
+            if (it->second.size() >= _output_max_num) {
+                slog::warn << "The different output size exceeds the maximum for image:" << input_image << slog::endl;
+                return;
+            }
+        }
+        write_binary_file(input_image, outputs);
+    }
+}
+
+/**
+ * @brief      Compare output with previous output
+ * @param[in]  inputimage Input image for inference request
+ * @param[in]  outputs Output tensor for inference request
+ * @return true if this output is different from previous output, otherwise false
+ */
+bool ResultDump::compare_output(const std::string& input_image, const ov::TensorVector& outputs) {
+    auto iter = _different_outputs.find(input_image);
+    if (iter == _different_outputs.end()) {
+        _different_outputs[input_image].reserve(_output_max_num);
+        return true;
+    }
+
+    auto output_size = outputs.size();
+    bool different = true;
+    for (auto& exist_outputs : iter->second) {
+        size_t offset = 0;
+        auto old_output = (char*)exist_outputs.buffer;
+        bool same = true;
+        for (size_t i = 0; i < output_size; i++) {
+            auto buffsize = outputs[i].get_byte_size();
+            auto new_output = outputs[i].data();
+            if (offset + buffsize > exist_outputs.size) {
+                slog::err << "New outputs size is different with the exist outputs size" << slog::endl;
+                same = false;
+                break;
+            } else {
+                if (0 != std::memcmp(new_output, old_output, buffsize)) {
+                    same = false;
+                    break;
+                }
+            }
+            offset += buffsize;
+            old_output += buffsize;
+        }
+
+        if (same) {
```

Comment:
Can we merge variable "same" and "different" to be one?

Reply:
I will replace "break" with "return" and delete the variable "different".